### ISSUE TITLE
[props-table] update react-docgen dependency

### ISSUE THREAD

**Issue by emilyrohrbough** [cerner/terra-core#2425]
# Feature Request

## Description
Update terra-props-table to use the current version of [`react-docgen`](https://github.com/reactjs/react-docgen). The current version pulls in babel-runtime which relies on babel v6

**Comment by emilyrohrbough** [comment]
Additionally, the current version of react-docgen is pulling in `recast` which is pulling in `core-js` v2. Terra-toolkit requires `core-js` v3. Bumping the version should also resolve this.

**Comment by avinashg1994** [comment]
I had updated the **react-docgen dependency** for **terra-props-table** from v2.15.0 to the latest stable release v.4.1.1. But on trying to start the local server, it failed to compile showing the following errors and warnings.
<img width="960" alt="react-docgen" src="https://user-images.githubusercontent.com/50138162/59345035-e7e2f900-8d2c-11e9-8c73-6991e04844db.png">
The same issue does not occur prior to updating the react-docgen version.

**Comment by emilyrohrbough** [comment]
@avinashg1994  Did you run `clean:install` after updating the react-docgen dependency? Was there any unmet peerDependencies for `fs` in the logs after installing?

**Comment by avinashg1994** [comment]
@emilyrohrbough, Yes i did run `clean:install` after the update. The error only occurs with the updated react-docgen version.
As for the peerDependencies for `fs`, i went through the logs after installation and found no errors or warnings, the installation logs were similar to the logs prior to the change.
I suspect the reason behind this issue is because of the multiple versions of source-map found, as seen in the warning. I also tried to locally change the terra-toolkit scripts to update the source-map version but was unable to do so.

**Comment by emilyrohrbough** [comment]
This is interesting. There appears to be a few bug logged around this:
- https://github.com/reactjs/react-docgen/issues/363
- https://github.com/thlorenz/convert-source-map/issues/64

@mhemesath recently discovered this module is quite heavy and is one of the largest in our webpack bundle. I wonder if it would be better to work on moving this generation to a pre-build step instead up updating.

**Comment by bjankord** [comment]
Let's hold of on doing this work for now. It looks like react-docgen v5 is in beta, I'd like to wait until that comes up.

We talked in scrum that when we want to look at move from using react-docgen at run-time to using it at build time to cut down on the webpack bundle.

We also talked about with doing this, we'll likely not add any more enhancements / dependencies updates to terra-props-table and plan to deprecate it once we add the ability to use react-docgen at build time.
